<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
 <head>
  <title>ErrorException</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
 </head>
 <body><div style="text-align: center;">
 <div class="prev" style="float: left;"><a href="reserved.exceptions.html">Предопределённые исключения</a></div>
 <div class="next" style="float: right;"><a href="errorexception.construct.html">ErrorException::__construct</a></div>
 <div class="up"><a href="reserved.exceptions.html">Предопределённые исключения</a></div>
 <div class="home"><a href="index.html">PHP Manual</a></div>
</div><hr /><div>
 <h1 class="title">ErrorException</h1>
 
 <div class="partintro">
 

  <div id="errorexception.intro" class="section">
   <h2 class="title">Введение</h2>
   <p class="para">
    Исключение в случае ошибки. 
   </p>
  </div>

 
  <div id="errorexception.synopsis" class="section">
   <h2 class="title">Class synopsis</h2>
 

   <div class="classsynopsis">
    <div class="ooclass"><b class="classname">ErrorException</b></div>
 

    <div class="classsynopsisinfo">
     <span class="ooclass">
       class <b class="classname">ErrorException</b>
     </span>
 
     <span class="ooclass">
      <span class="modifier">extends</span>
      <b class="classname"> Exception</b>
     </span>
     {</div>

 
    <div class="classsynopsisinfo classsynopsisinfo_comment">/* Свойства */</div>
    <div class="fieldsynopsis">
     <span class="modifier">protected</span>
     <span class="type">int</span>
     <var class="varname"><a href="class.errorexception.html#errorexception.props.severity">$severity</a></var>
    ;</div>

 
    <div class="classsynopsisinfo classsynopsisinfo_comment">/* Методы */</div>
    <div class="constructorsynopsis dc-description">
   <span class="modifier">public</span>  <span class="methodname"><b><a href="errorexception.construct.html" class="function">__construct</a></b></span>
    ([ <span class="methodparam"><span class="type">string</span> <tt class="parameter">$message</tt></span>
   [, <span class="methodparam"><span class="type">int</span> <tt class="parameter">$code</tt></span>
   [, <span class="methodparam"><span class="type">int</span> <tt class="parameter">$severity</tt></span>
   [, <span class="methodparam"><span class="type">string</span> <tt class="parameter">$filename</tt></span>
   [, <span class="methodparam"><span class="type">int</span> <tt class="parameter">$lineno</tt></span>
  ]]]]] )</div>

    <div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">int</span> <span class="methodname"><b><a href="errorexception.getseverity.html" class="function">getSeverity</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>

 
    <div class="classsynopsisinfo classsynopsisinfo_comment">/* Унаследованные методы */</div>
    <div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><b><a href="exception.getmessage.html" class="function">Exception::getMessage</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">int</span> <span class="methodname"><b><a href="exception.getcode.html" class="function">Exception::getCode</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><b><a href="exception.getfile.html" class="function">Exception::getFile</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><b><a href="exception.getline.html" class="function">Exception::getLine</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">array</span> <span class="methodname"><b><a href="exception.gettrace.html" class="function">Exception::getTrace</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><b><a href="exception.gettraceasstring.html" class="function">Exception::getTraceAsString</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">public</span> <span class="type">string</span>  <span class="methodname"><b><a href="exception.tostring.html" class="function">Exception::__toString</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">private</span> <span class="type">string</span> <span class="methodname"><b><a href="exception.clone.html" class="function">Exception::__clone</a></b></span>
    ( <span class="methodparam">void</span>
   )</div>

   }</div>

 
  </div>
 

  <div id="errorexception.props" class="section">
   <h2 class="title">Свойства</h2>
   <dl>

    <dt id="errorexception.props.severity">
     <span class="term"><var class="varname">severity</var></span>

     <dd>

      <p class="para">Важность ошибки</p>
     </dd>

    </dt>

   </dl>

  </div>


  <div id="errorexception.examples" class="section">
   <h2 class="title">Примеры</h2>
   <p class="para">
    <div class="example" id="errorexception.example.error-handler" name="errorexception.example.error-handler">
     <p><b>Пример #1 Пример использования <span class="classname">ErrorException</span></b></p>
     <div class="example-contents"><p>
      Превращаем все ошибки в <span class="classname">ErrorException</span>.
     </p></div>
     <div class="example-contents">
 <div class="phpcode"><code><span style="color: #000000">
<span style="color: #0000BB">&lt;?php<br /></span><span style="color: #007700">function&nbsp;</span><span style="color: #0000BB">exception_error_handler</span><span style="color: #007700">(</span><span style="color: #0000BB">$errno</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">$errstr</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">$errfile</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">$errline&nbsp;</span><span style="color: #007700">)&nbsp;{<br />throw&nbsp;new&nbsp;</span><span style="color: #0000BB">ErrorException</span><span style="color: #007700">(</span><span style="color: #0000BB">$errstr</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">0</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">$errno</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">$errfile</span><span style="color: #007700">,&nbsp;</span><span style="color: #0000BB">$errline</span><span style="color: #007700">);<br />}<br /></span><span style="color: #0000BB">set_error_handler</span><span style="color: #007700">(</span><span style="color: #DD0000">"exception_error_handler"</span><span style="color: #007700">);<br /><br /></span><span style="color: #FF8000">/*&nbsp;вызываем&nbsp;ошибку&nbsp;*/<br /></span><span style="color: #0000BB">strpos</span><span style="color: #007700">();<br /></span><span style="color: #0000BB">?&gt;</span>
</span>
</code></div>
     </div>

     <div class="example-contents"><p>Результатом выполнения данного примера
будет что-то подобное:</p></div>
     <div class="example-contents"><pre>
<div class="cdata"><pre>
Fatal error: Uncaught exception &#039;ErrorException&#039; with message &#039;Wrong parameter count for strpos()&#039; in /home/bjori/tmp/ex.php:8
Stack trace:
#0 [internal function]: exception_error_handler(2, &#039;Wrong parameter...&#039;, &#039;/home/bjori/php...&#039;, 8, Array)
#1 /home/bjori/php/cleandocs/test.php(8): strpos()
#2 {main}
  thrown in /home/bjori/tmp/ex.php on line 8
</pre></div>
     </pre></div>
    </div>
   </p>
  </div>
 
 </div>
 
 



 



 



 



 
<h2>Содержание</h2><ul class="chunklist chunklist_reference"><li><a href="errorexception.construct.html">ErrorException::__construct</a> — Construct the exception</li><li><a href="errorexception.getseverity.html">ErrorException::getSeverity</a> — Gets the exception severity</li></ul>
</div>
<hr /><div style="text-align: center;">
 <div class="prev" style="float: left;"><a href="reserved.exceptions.html">Предопределённые исключения</a></div>
 <div class="next" style="float: right;"><a href="errorexception.construct.html">ErrorException::__construct</a></div>
 <div class="up"><a href="reserved.exceptions.html">Предопределённые исключения</a></div>
 <div class="home"><a href="index.html">PHP Manual</a></div>
</div></body></html>
